### DISCUSSION TITLE
widget.NewSelect() Api enhancement request

### DISCUSSION THREAD

**Discussion by hq-zhonger** [fyne-io/fyne#3667] [Q&A] [answered]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

<img width="1120" alt="捕获" src="https://user-images.githubusercontent.com/78897251/219677329-95f116a7-7e7d-478e-99ac-af87865d9a1f.PNG">

When selecting the right selection box, refresh the contents of the other selection box according to the contents of the selection box

### Describe the solution you'd like to see.

package main

import "fyne.io/fyne/v2/widget"

type App struct {
	key    *widget.Select
	values *widget.Select
}

func (MyApp *App) ReturnKeyStringList() []string {
	return []string{
		"ThinkPHP",
		"WordPress",
	}
}

func (MyApp *App) ReturnThinkPHPStringList() []string {
	return []string{
		"ThinkPHP Sql",
		"ThinkPHP Xss",
	}
}

func (MyApp *App) ReturnWordPressStringList() []string {
	return []string{
		"WordPress Sql",
		"WordPress Ssrf",
	}
}

func main() {
	var MyApp App

	MyApp.key = widget.NewSelect(MyApp.ReturnKeyStringList(), func(s string) {
		switch s {
		case "ThinkPHP":
			MyApp.values.SetSelectStringList(MyApp.ReturnThinkPHPStringList())
			break
		case "WordPress":
			MyApp.values.SetSelectStringList(MyApp.ReturnWordPressStringList())
			break
		default:
			break
		}
	})

	MyApp.key.SetSelectedIndex(0)

	MyApp.values = widget.NewSelect([]string{""}, func(s string) {

	})

	MyApp.values.SetSelectedIndex(0)
}

**Comment by Jacalz** [comment, marked as answer]
This shouldn’t require any API additions as far as I am aware. All you need to do is set the `OnChanged` method of one widget and make sure that it sets the contents of the other one. Just make sure that you don’t have an infinite loop of them updating each other.
